From 9a3aa2c4df257a0b223d23b7e3481f2e257a3bca Mon Sep 17 00:00:00 2001
From: Hai Pham <hai.pham.ud@renesas.com>
Date: Sat, 26 Apr 2025 14:56:55 +0700
Subject: [PATCH 15/52] arm64: renesas: Extend stub PSCI implementation to
 R-Car Gen5

Extend the support to R-Car Gen5, with slightly different SoC reset
functionality from R-Car Gen4.

Signed-off-by: Hai Pham <hai.pham.ud@renesas.com>
---
 arch/arm/mach-renesas/psci-rcar64.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/mach-renesas/psci-rcar64.c b/arch/arm/mach-renesas/psci-rcar64.c
index 459dd55ff45..22c2ee045cc 100644
--- a/arch/arm/mach-renesas/psci-rcar64.c
+++ b/arch/arm/mach-renesas/psci-rcar64.c
@@ -32,7 +32,12 @@ u32 __secure psci_version(void)
 
 void __secure __noreturn psci_system_reset(void)
 {
+#if defined(CONFIG_RCAR_GEN5)
+	writel(RST_KCPROT_DIS, RST_RESKCPROT0);
+	writel(0x1, RST_SWSRES1A);
+#else
 	writel(RST_SPRES, RST_SRESCR0);
+#endif
 
 	while (1)
 		;
-- 
2.34.1

